# Lab - Webscraping Using Python

In this lab, you are required to write python code to open a webpage and scrape the information on the website using BeautifulSoup.

Follow these instructions:
<ul>
<li>Write python code with BeautifulSoup and requests modules</li>
<li> Part 1: Read the html contents from https://towardsdatascience.com/deploying-a-python-web-app-on-aws-57ed772b2319</li>
<li>Part 2: Get the table data from "https://archive.ics.uci.edu/ml/datasets.php?format=&task=&att=&area=&numAtt=&numIns=&type=&sort=nameUp&view=table"
</ul>


In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

## Part 1 

In [2]:
#Write your code below

# Given the url below get the html content of the site
url = "https://towardsdatascience.com/deploying-a-python-web-app-on-aws-57ed772b2319"
html = "####"
# Use Beautiful soup to generate the soup term. You are essentially converting
# the html content to a beautiful soup variable
soup = "####"
soup_type = "####"
print("The type of soup variable is {}".format(soup_type))

The type of soup variable is <class 'bs4.BeautifulSoup'>


The type of soup variable is <class 'bs4.BeautifulSoup'>


In [3]:

# Get the title of the article. You will have to go to url and check which div contains the title header
tag = "####"
article_title = tag.text 

# Get all the image locations/src.
# The first step is to find the article tag. This is because we only care about the images in the artilce
article_tag = "####"

# Next find all the image tags
img_tags = "####"

# We will store all the image links in this variable
img_src = []

# Write a for loop that checks is src is in the image tag attributes and if so appends src to img_src
for img in img_tags: 
    if"####" in "####":
        "####"

len_images = "####"
print("Title of the article is: {}".format(article_title))

print("Total number of images in the article: {}".format(len_images))

Title of the article is: Deploying a Python Web App on AWS
Total number of images in the article: 17


Title of the article is: Deploying a Python Web App on AWS <br>
Total number of images in the article: 17

## Part 2 

In [4]:
# Scrape the UCI table data and get the soup variable the same way you did in part 1   
uci_url = "https://archive.ics.uci.edu/ml/datasets.php?format=&task=&att=&area=&numAtt=&numIns=&type=&sort=nameUp&view=table"
html = "####"
soup = "####"


In [5]:
# You will need to get the table tag. There are two tables in this url, you will need the first table
table_tag = "####"

# This tells you how many rows of data are in the table
rows = len(table_tag)-1

# In this part we will go through the table tags get and the text from each row. Remember to split the text 
# to store in a list. The variable store_rows will be a list of lists. 
store_rows = [] 
for t in table_tag: 
        "####"    

# Covert store rows to a dataframe        
df= "####"

# Rename the columns of df to the first row of the dataset 
"####" = "####"

# Drop the first row of the dataset
df = "####"

# Get the head of the dataset
df_head = "####"


print("Number of rows of data: {}\n".format(rows))
print("Head of the datset is:\n{}".format(df_head))

Number of rows of data: 153

Head of the datset is:
0                                               Name    \
1                2.4 GHZ Indoor Channel Measurements     
2           3D Road Network (North Jutland, Denmark)     
3                                         3W dataset     
4       : Simulated Data set of Iraqi tourism places     
5     A study of  Asian Religious and Biblical Texts     

0                  Data Types                 Default Task Attribute Types  \
1               Multivariate               Classification            Real    
2           Sequential, Text       Regression, Clustering            Real    
3  Multivariate, Time-Series   Classification, Clustering   Integer, Real    
4               Multivariate   Classification, Clustering                    
5         Multivariate, Text   Classification, Clustering         Integer    

0 # Instances # Attributes   Year      
1       7840            5   2018       
2     434874            4   2013       
3       19

Number of rows of data: 153 <br>
<br>
0                                               Name    \ <br>
1                2.4 GHZ Indoor Channel Measurements     <br>
2           3D Road Network (North Jutland, Denmark)     <br>
3                                         3W dataset     <br>
4       : Simulated Data set of Iraqi tourism places     <br>
5     A study of  Asian Religious and Biblical Texts     <br>
<br>
0                  Data Types                 Default Task Attribute Types  \ <br>
1               Multivariate               Classification            Real     <br>
2           Sequential, Text       Regression, Clustering            Real     <br>
3  Multivariate, Time-Series   Classification, Clustering   Integer, Real     <br>
4               Multivariate   Classification, Clustering                    <br>
5         Multivariate, Text   Classification, Clustering         Integer     <br>
<br>
0 # Instances # Attributes   Year      <br>
1       7840            5   2018       <br>
2     434874            4   2013       <br>
3       1984            8   2019       <br>
4        232           16   2020       <br>
5        590         8265   2019    <br>